python3Packages: patch updates

FRidh opened the pull request:
###### Description of changes

Reference https://hydra.nixos.org/eval/1784131


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

lourkeur commented:
fix build for chiapos: https://github.com/NixOS/nixpkgs/pull/199417/files#diff-30e6c1dab67b0202ec035585ff202ccf40b0ca516ecdb3dc1cdc0151dc5be833

(I will look at chiavdf asaic)

lourkeur commented:
#199889

kirillrdy commented:
fixes tabulate and awsebcli https://github.com/NixOS/nixpkgs/pull/199470

raboof commented:
Looks like this PR gained some merge conflicts - would it be helpful if I tried to rebase or would you prefer to keep the lead here yourself?

lourkeur commented:
two more ood packages #201869

FRidh commented:
Unfortunately I did not have the time to finish this. Still, some commits can probably be cherry-picked.

mweinelt committed:
python39Packages: Stop recursing into package set

For NixOS 23.05 we are shifting support away from 3.9 and towards 3.10
and 3.11.

mweinelt committed:
python311Packages: Recurse into package set

Python 3.11 is the latest stable release that we are going to support,
together with 3.10, in NixOS 23.05. The default python version will
remain at 3.10, but 3.11 packages will be built on Hydra and provided
from the binary cache.
